Moloco Native ad support in adapter.

vishdhmnmoloco opened the pull request:
Sample PR to add Native ads suport to Google Ads Moloco Adapter.
Supports Moloco SDK 3.6.0 and above.

- This change adds support for Moloco adapter to have Native Ad support.
- The adapter is not tested by Moloco as native ads is not enabled
- Please do the verification before merging this change in. I had trouble establishing the build dependencies and prepared the adapter from our internal source without being able to compile the google project.

thanvirm commented:
CLA check passes now. Yay!

But, looks like there is a conflict. Could you resolve the conflict? Thanks!

thanvirm commented:
Hey Vishal,

Looks like there are conflicts in NativeAdLoader.swift with the code in the main branch. Could you resolve those conflicts? Thanks!

vishdhmnmoloco commented:
Hi @thanvirm ,

I have rebased the changes to top of main and made necessary changes to resolve conflicts.
Also updated the tests to use the latest changes, please let me know if there is anything else I can help with.

Thanks!

thanvirm commented on AdapterUnitTests/MolocoAdapterTests/MolocoNativeAdTest.swift:
```diff
@@ -0,0 +1,105 @@
+  /// A bid response received by the adapter to load the ad.
+  static let testBidResponse = "bid_response"
+
+  func testNativeLoadSuccess() {
```
In this test, could you also verify that Moloco's native ad assets are correctly mapped to Google's native ad assets? Similar to the tests here: https://github.com/googleads/googleads-mobile-ios-mediation/blob/c64cabd74930e59a31980e47b6ccec5fe1586928/AdapterUnitTests/PangleAdapterTests/AUTPangleNativeAdTests.m#L168-L234 

In the linked example, every mapping is a separate test. You can also verify all those in a single test (as part of testNativeLoadSuccess).

Thanks!

vishdhmnmoloco replied:
Hi @thanvirm I created a new PR with this change. It also uncovered a Swift Double to NSDecimalNumber conversion issue, that is also fixed in this PR.

https://github.com/googleads/googleads-mobile-ios-mediation/pull/481/files
